src/Actions/helpers/Web3ActionCreator.js

```diff
@@ -14,29 +14,14 @@
 //   return { type: WEB3_REQUEST_SUCCEEDED, name, result }
 // }
 
-export function web3Request(name, args, provider, next) {
-  // This specifically pulls state from the web3 reducer. Smell?
-  let web3 = new Web3(provider)
-  
-  //dispatch(Web3RequestStarted(name))
-
-  let fn = web3.eth[name]
-
-  args.push(next)
+export async function web3Request(name, args, web3Instance) {
+  let fn = web3Instance.eth[name]
 
-  fn.apply(web3.eth, args)
+  return await fn.apply(web3Instance.eth, args)
 }
 
-export function web3ActionCreator(name, args, next) {
-  if (typeof args == "function") {
-    next = args
-    args = []
-  }
-
-  return function(dispatch, getState) {
-    let provider = getState().web3.provider
-    web3Request(name, args, provider, (err, result) => {
-      next(result, dispatch, getState)
-    })
-  }
-}
\ No newline at end of file
+export async function web3ActionCreator(dispatch, getState, name, args) {
+  // This specifically pulls state from the web3 reducer. Smell?
+  let web3Instance = getState().web3.web3Instance
+  return await web3Request(name, args, web3Instance)
+}
```

Comment:
My guess is that the changes in this file are likely to be the most contentious. Given that we're not dispatching/handling the `WEB3_REQUEST_*` actions noted above, I don't think there's any value in this very thin layer of indirection.